### ISSUE TITLE
err catching is too aggressive. holochain sometimes has non-fatal errors that are logged to stderr, such as lair-keystore

### ISSUE THREAD

**Issue by Connoropolous** [holochain/tryorama#119]
https://github.com/holochain/tryorama/blob/20409984b18184978b644c7a44c3b8901c50f118/ts/src/local/conductor.ts#L220-L222

**Comment by Connoropolous** [comment]
ping @jost-s

**Comment by jost-s** [comment]
@Connoropolous It'd be great if you could provide the error message that causes the rejection.

**Comment by Connoropolous** [comment]
sure: 

```
error: error when starting conductor: Jun 12 18:27:36.963 ERROR lair_keystore_client: error=IpcClientConnectError("/private/var/folders/2y/ydm_32156jn_t6lm4vymkt6w0000gn/T/B-hYDTVIYJEa6q6O61u0M/keystore/socket", Os { code: 2, kind: NotFound, message: "No such file or directory" }) file="/Users/x/.cargo/registry/src/github.com-1ecc6299db9ec823/lair_keystore_client-0.0.10/src/lib.rs" line=41
```

**Comment by jost-s** [comment]
What is this error about, do you know?

When starting the conductor and met with an error, it could just output the error without rejecting the promise. Then the startup is only unsuccessful when the underlying command `hc` encounters a fatal error. Or do you have something else in mind?

**Comment by Connoropolous** [comment]
that's what I've done by just diving into my node_modules folder and modifying it, yeah. 

As far as I've seen, that error pops up anytime starting holochain, unless the lair_keystore process is already running in advance. if it doesn't see it running there, then it starts it (the lair_keystore process)

**Comment by jost-s** [comment]
> As far as I've seen, that error pops up anytime starting holochain, unless the lair_keystore process is already running in advance. if it doesn't see it running there (in the configured directory as an IPC process it can connect to), then it starts it (the lair_keystore process)

Yes, but for me and on the CI it doesn't keep the conductor from starting. It's odd that it's happening on your machine.

**Comment by Connoropolous** [comment]
NOT in nix-shell

Apple M1 Max
MacOS Monterey version 12.4

```
hc --version
holochain_cli 0.0.41
```

```
holochain --version
holochain 0.0.143
```

**Comment by Connoropolous** [comment]
It occurs to me the lair version could be the source of difference

**Comment by jost-s** [comment]
most likely I think. Is it v0.1.x?

**Comment by Connoropolous** [comment]
no mine is 
```
lair-keystore --version
lair-keystore 0.0.10
```
what would yours show if you were in nix-shell with those holochain versions?

**Comment by jost-s** [comment]
It's identical

```bash
$ lair-keystore --version
lair-keystore 0.0.10
```

🤔

**Comment by jost-s** [comment]
> most likely I think. Is it v0.1.x?

Sorry, you already wrote that in your previous post 🤭

**Comment by Connoropolous** [comment]
that increases the likelihood of it being a nix-shell vs non nix-shell thing I guess

**Comment by Connoropolous** [comment]
I have confirmed that it also occurs within nix-shell. 

I don't know why but normal logs are being put to `stderr` instead of stdout. The failure of my CI for hrea doesn't have good logs, but it is technically evidence of this.
https://github.com/h-REA/hREA/runs/6889841102?check_suite_focus=true

that branch at the related commit is fine for steps-to-reproduce. 

I need to get this up and running, I think I might quickly fork it and push my fix, and you can see what makes sense to you to do with this info?

**Comment by jost-s** [comment]
Running the tests locally, I always run into the same compilation error
```bash
Compiling hc_zome_agent_registration_lib v0.1.0 (https://github.com/holochain-open-dev/agent-registration?branch=hdk-123-validation#c29b1e93)
error[E0412]: cannot find type `IdentRemote` in this scope
   --> lib/hdk_uuid_types/src/lib.rs:203:23
    |
203 |         let external: IdentRemote = (DnaHash::from_raw_36(vec![0xdb; HOLO_HASH_UNTYPED_LEN]), wrapped).into();
    |                       ^^^^^^^^^^^ not found in this scope

error[E0277]: the trait bound `HoloHash<AnyDht>: From<SomeValue>` is not satisfied
   --> lib/hdk_uuid_types/src/lib.rs:96:14
    |
96  |         impl $crate::DnaAddressable<$base> for $r {
    |              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ the trait `From<SomeValue>` is not implemented for `HoloHash<AnyDht>`
...
199 |         addressable_identifier!(Ident => SomeValue);
    |         ------------------------------------------- in this macro invocation
    |
    = help: the following implementations were found:
              <HoloHash<AnyDht> as From<HoloHash<AnyLinkable>>>
              <HoloHash<AnyDht> as From<HoloHash<holo_hash::hash_type::Agent>>>
              <HoloHash<AnyDht> as From<HoloHash<holo_hash::hash_type::Entry>>>
              <HoloHash<AnyDht> as From<HoloHash<holo_hash::hash_type::Header>>>
            and 11 others
note: required by a bound in `DnaAddressable`
   --> lib/hdk_uuid_types/src/lib.rs:68:21
    |
63  | pub trait DnaAddressable<B>
    |           -------------- required by a bound in this
...
68  |         AnyDhtHash: From<B>,
    |                     ^^^^^^^ required by this bound in `DnaAddressable`
    = note: this error originates in the macro `addressable_identifier` (in Nightly builds, run with -Z macro-backtrace for more info)
```

**Comment by jost-s** [comment]
> I need to get this up and running, I think I might quickly fork it and push my fix, and you can see what makes sense to you to do with this info?

I'll remove the promise rejection. In any case I think that the `hc` command should handle conductor errors, not Tryorama. Will do that now.

**Comment by Connoropolous** [comment]
Let me know if you are definitely wanting steps to reproduce. Sounds like taking out the reject is good for now. The logs will still look weird though, since things will be logged looking like errors that aren't

**Comment by jost-s** [comment]
> since things will be logged looking like errors that aren't

The error that the keystore isn't found at conductor startup is an error IMO. It's way peculiar that `hc` is written to compensate for the error, instead of fixing the error at the origin. (that'll happen with keystore 0.1.x)

In any case I think the mechanism to log errors should be kept in Tryorama.

It would be nice to know why that error occurs, but if it's a bigger effort, then I'll let go of it.

**Comment by jost-s** [comment]
fixed by #122
